File: java/src/main/java/pl/adambanaszkiewicz/four_layers_example/domain_layer/from_this/Product.java
package pl.adambanaszkiewicz.four_layers_example.domain_layer.from_this;

import jakarta.persistence.*;

@Entity
public class Product {
    @Id
    private String id;
    private int qty;
    private double price;

    public Product(String id, int qty, double price) {
        this.id = id;
        this.qty = qty;
        this.price = price;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public int getQty() {
        return qty;
    }

    public void setQty(int qty) {
        this.qty = qty;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }
}

File: php/src/DomainLayer/FromThis/Product.php
<?php

declare(strict_types=1);

namespace App\DomainLayer\FromThis;

final class Product
{
    private string $id;

    public function __construct(
        private string $productId,
        private int $qty,
        private float $price,
    ) {
    }

    public function setQty(int $qty): void
    {
        $this->qty = $qty;
    }

    public function getQty(): int
    {
        return $this->qty;
    }

    public function getProductId(): string
    {
        return $this->productId;
    }

    public function setProductId(string $productId): void
    {
        $this->productId = $productId;
    }

    public function getPrice(): float
    {
        return $this->price;
    }

    public function setPrice(float $price): void
    {
        $this->price = $price;
    }
}
